pyccel/codegen/printing/fcode.py

```diff
@@ -2436,9 +2436,9 @@ def _print_LiteralFloat(self, expr):
         return "%s_C_DOUBLE" % printed
 
     def _print_LiteralComplex(self, expr):
-        real_str = self._print_Float(expr.real)
-        imag_str = self._print_Float(expr.imag)
-        return "({}, {})".format(real_str, imag_str)
+        real_str = self._print(expr.real)
+        imag_str = self._print(expr.imag)
+        return "cmplx({}, {}, kind = {})".format(real_str, imag_str, expr.precision)
```

Comment:
I don't understand why we need to call the `cmplx` function in order to print a complex literal constant. `expr.real` and `expr.imag` should inherit the correct precision from `expr`, and hence they should be printed correctly. For example
```fortran
(1.0_C_DOUBLE, 2.5_C_DOUBLE)
```

Reply:
it was @EmilyBourne  who made that change I think she had some reason to do that.

Reply:
This was causing problems when a function calls another and passes a complex literal constant. I have checked out the branch at the point at which I had the problem. It was caused by :
```
z = gen_6(1j, 1j)
```
for 
```
#$ header template S(int|real|complex|bool)
@types('S', 'S')
def gen_6(x, y):
    return x + y
```

The code generated was:
```
  z = gen_6((1_C_LONG_LONG, 2_C_LONG_LONG),
      (1_C_LONG_LONG, 2_C_LONG_LONG))
```
which generated the errors:
```
generic_functions.f90:562:44:

   z = gen_6((1_C_LONG_LONG, 2_C_LONG_LONG),
                                            1
Error: Syntax error in argument list at (1)
generic_functions.f90:563:6:

       (1_C_LONG_LONG, 2_C_LONG_LONG))
      1
Error: Unclassifiable statement at (1)
```

It is really due to an underlying problem which is that we don't check that the arguments are `LiteralFloat` when we create a `LiteralComplex`. Presumably one had been missed. This seems to have been corrected elsewhere though as I cannot reproduce the problem on the branch in its current state. I am therefore happy to revert these changes

Reply:
I have opened PR #572 to ensure Literals are correctly handled